app.py

```diff
@@ -79,31 +132,41 @@ def post_to_discord(channel, text):
              json={"content": text},
              headers={"Authorization": "Bot " + config["token"],
                       "User-Agent": "gitlab-discord-bridge by blha303"
-                     }).json()
+                      }).json()
     return "cool"
 
-@app.route('/<channelid>', methods=['GET', 'POST'])
+
+@app.route('/<channelid>', methods=['POST'])
 def index(channelid):
     if request.method != "POST":
-        return make_response("lol", 400)
-    if ("host" in config and request.remote_addr != config["host"]) or request.headers.get("X-Gitlab-Token", "") != config["secret"]:
-        return make_response("go away please", 403)
+        return make_response("Method Not Allowed", 405)
+    if ("host" in config and request.remote_addr != config["host"]) or request.headers.get("X-Gitlab-Token", "") != \
+            config["secret"]:
+        return make_response("Not Authorized", 403)
     try:
         body = request.get_json()
     except BadRequest:
-        return make_response("lol", 400)
+        return make_response("Bad Request", 400)
 
     handlers = {"push": handle_push,
                 "tag_push": handle_tag,
                 "issue": handle_issue,
                 "note": handle_note,
                 "merge_request": handle_merge,
                 "wiki_page": handle_wiki,
-                "pipeline": handle_pipeline
-               }
+                "pipeline": handle_pipeline,
+                "build": handle_build
+                }
+    try:
+        if body["object_kind"] in handlers and body["object_kind"] in config["enabled_hooks"]:
+            return make_response(post_to_discord(channelid, handlers[body["object_kind"]](body)), 200)
+    except KeyError:
+        if body["object_kind"] in handlers and "enabled_hooks" not in config:
+            return make_response(post_to_discord(channelid, handlers[body["object_kind"]](body)), 200)
```

Comment:
A message should be printed to inform the user if the config needs to be updated.

Reply:
Ok lets have a discussion here, you are correct that enabled_hooks are not necessary as that can be configured in gitlab, so should I just remove this entire piece of code and revert it to the original one without the enabled_hooks checks?

Reply:
I'd say so, yeah. I'd like to avoid any breaking changes with anyone that tries to update the program without changing their config. To that end I'd probably remove the assert lines at the top and change `app.run` to use `config.get("listen", "0.0.0.0")` and `config.get("port", 25431)` instead. The only required lines in the config should be `secret` and `token` ideally.

Reply:
Already removed the additional config and the startup does not fail when listen or port is not set up. But using config.get with the default is better than my ifs:). Going to change this once I get home.